src/modules/rdf/functions.cpp

```diff
@@ -224,6 +224,46 @@ bool RDFModule::calculateGRCells(ProcessPool &procPool, Configuration *cfg, Part
  * Public Functions
  */
 
+// Calculate and return effective density for based on the target Configurations
+double RDFModule::effectiveDensity() const
+{
+    auto rho0 = 0.0, totalWeight = 0.0;
+    for (auto *cfg : targetConfigurations())
+    {
+        // TODO Get weight for configuration
+        auto weight = 1.0;
+
+        totalWeight += weight;
+
+        rho0 += weight / cfg->atomicDensity();
+    }
+    rho0 /= totalWeight;
+    rho0 = 1.0 / rho0;
```

Comment:
```suggestion
    auto [totalWeight, rho0] = std::accumulate(targetConfigurations().begin(), targetConfigurations().end(), std::make_tuple(0.0, 0.0), [](auto acc, auto *cfg) {
        // TODO Get weight for configuration
        auto weight = 1.0;
        auto [accWeight, accRho] = acc;
        return std::make_tuple(accWeight + weight, accRho + weight/cfg->atomicDensity());
    });

    rho0 /= totalWeight;
    rho0 = 1.0 / rho0;
```

I *think* this should work.  The tuple is a little unfortunate, but it saves us having to calculate the weight twice (when we start calculating the weight).

Reply:
Happy to try it out - in the interests of getting the backlog of PRs in following the OSX failures, I'll address this in a future bugfix.